src/reducers/project-id.js

```diff
@@ -1,27 +1,380 @@
+            return Object.assign({}, state, {
+                projectState: ProjectState.SAVING_WITH_ID_BEFORE_NEW
+            });
+        }
+        return state;
+    case GO_TO_ERROR_STATE:
+    // NOTE: we should introduce handling in components for showing ERROR state
+        if ([
+            ProjectState.NOT_LOADED,
+            ProjectState.FETCHING_WITH_ID,
+            ProjectState.FETCHING_NEW_DEFAULT,
+            ProjectState.FETCHING_NEW_DEFAULT_TO_SAVE
+        ].includes(state.projectState)) {
+            return Object.assign({}, state, {
+                projectState: ProjectState.ERROR,
+                errStr: action.errStr
+            });
+        }
+        return state;
     default:
         return state;
     }
 };
 
-const setProjectId = function (id) {
-    return {
-        type: SET_PROJECT_ID,
-        id: id
-    };
+const doneCreatingNew = id => ({
+    type: DONE_CREATING_NEW,
+    id: id
+});
+
+const doneFetchingProjectData = (projectData, projectState) => {
+    switch (projectState) {
+    case ProjectState.FETCHING_WITH_ID:
+        return {
+            type: DONE_FETCHING_PROJECT_DATA_WITH_ID,
+            projectData: projectData
+        };
+    case ProjectState.FETCHING_NEW_DEFAULT:
+        return {
+            type: DONE_FETCHING_DEFAULT_PROJECT_DATA,
+            projectData: projectData
+        };
+    case ProjectState.FETCHING_NEW_DEFAULT_TO_SAVE:
+        return {
+            type: DONE_FETCHING_DEFAULT_PROJECT_DATA_TO_SAVE,
+            projectData: projectData
+        };
+    default:
+        break;
+    }
+};
+
+const doneLoadingFileUpload = () => ({
+    type: DONE_LOADING_VM_FILE_UPLOAD
+});
+
+const doneLoading = projectState => {
+    switch (projectState) {
+    case ProjectState.LOADING_VM_WITH_ID:
+        return {
+            type: DONE_LOADING_VM_WITH_ID
+        };
+    case ProjectState.LOADING_VM_FILE_UPLOAD:
+        return {
+            type: DONE_LOADING_VM_FILE_UPLOAD
+        };
+    case ProjectState.LOADING_VM_NEW_DEFAULT:
+        return {
+            type: DONE_LOADING_VM_NEW_DEFAULT
+        };
+    case ProjectState.LOADING_VM_NEW_DEFAULT_TO_SAVE:
+        return {
+            type: DONE_LOADING_VM_NEW_DEFAULT_TO_SAVE
+        };
+    default:
+        break;
+    }
+};
+
+const doneSavingWithId = projectState => {
+    switch (projectState) {
+    case ProjectState.SAVING_WITH_ID:
+        return {
+            type: DONE_SAVING_WITH_ID
+        };
+    case ProjectState.SAVING_WITH_ID_BEFORE_NEW:
+        return {
+            type: DONE_SAVING_WITH_ID_BEFORE_NEW
+        };
+    default:
+        break;
+    }
 };
 
+const goToErrorState = errStr => ({
+    type: GO_TO_ERROR_STATE,
+    errStr: errStr
+});
+
+const setHashProjectId = id => ({
+    type: SET_HASH_PROJECT_ID,
+    id: id
+});
+
+const setInitialProjectId = id => ({
+    type: SET_INITIAL_PROJECT_ID,
+    id: id
+});
+
+const stepTowardsCreatingNewProject = () => ({
```

Comment:
I think this is confusingly verbose, because "step" towards seems to me like something that might be called multiple times until the project is created. But it looks like this is the initial action, so maybe `initiateCreateProject`?

Reply:
I'm using `newProjectRequested`. Sound good?